core/block/editor/userdataobject/userdata.go

```diff
@@ -0,0 +1,271 @@
+		SpaceID:  u.SpaceID(),
+	}, func(contactObject smartblock.SmartBlock) error {
+		state := contactObject.NewState()
+		for key, value := range contact.Details().Iterate() {
+			state.SetDetailAndBundledRelation(key, value)
+		}
+		return contactObject.Apply(state)
+	})
+	if err != nil {
+		log.Errorf("update contact: %v", err)
+	}
+}
+
+func (u *userDataObject) ListContacts(ctx context.Context) ([]*Contact, error) {
+	collection, err := u.state.Collection(ctx, ContactsCollection)
+	if err != nil {
+		return nil, fmt.Errorf("get collection: %w", err)
+	}
+	qry := collection.Find(nil)
+	contacts, err := u.queryContacts(ctx, qry)
+	if err != nil {
+		return nil, fmt.Errorf("query messages: %w", err)
+	}
+	return contacts, nil
+}
+
+func (u *userDataObject) queryContacts(ctx context.Context, query anystore.Query) ([]*Contact, error) {
+	iter, err := query.Iter(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("find iter: %w", err)
+	}
+	defer iter.Close()
+
+	var res []*Contact
+	for iter.Next() {
+		doc, err := iter.Doc()
+		if err != nil {
+			return nil, fmt.Errorf("get doc: %w", err)
+		}
+		message := NewContactFromJson(doc.Value())
+		res = append(res, message)
+	}
+	return res, nil
+}
+
+func (u *userDataObject) Close() error {
+	u.cancel()
+	return u.SmartBlock.Close()
+}
+
+func (u *userDataObject) SaveContact(ctx context.Context, profile *model.IdentityProfile, symKey crypto.SymKey) error {
+	arena := u.arenaPool.Get()
+	defer func() {
+		arena.Reset()
+		u.arenaPool.Put(arena)
+	}()
+	rawKey, err := symKey.Raw()
+	if err != nil {
+		return fmt.Errorf("get raw sym key: %w", err)
+	}
+	encodedKey := base64.StdEncoding.EncodeToString(rawKey)
+	contact := NewContact(profile.Identity, profile.Name, profile.Description, profile.IconCid, encodedKey, profile.GlobalName)
+	builder := &storestate.Builder{}
+	err = builder.Create(ContactsCollection, domain.NewContactId(profile.Identity), contact.ToJson(arena))
+	if err != nil {
+		return fmt.Errorf("create chat: %w", err)
+	}
+	_, err = u.storeSource.PushStoreChange(ctx, source.PushStoreChangeParams{
+		Changes: builder.ChangeSet,
+		State:   u.state,
+		Time:    time.Now(),
+	})
+	if err != nil {
+		return fmt.Errorf("push change: %w", err)
+	}
+	return nil
+}
+
+func (u *userDataObject) UpdateContactByIdentity(ctx context.Context, profile *model.IdentityProfile) error {
+	arena := u.arenaPool.Get()
+	defer func() {
+		arena.Reset()
+		u.arenaPool.Put(arena)
+	}()
+
+	builder := &storestate.Builder{}
+	err := u.updateContactFields(profile, builder)
+	if err != nil {
+		return err
+	}
+	_, err = u.storeSource.PushStoreChange(ctx, source.PushStoreChangeParams{
+		Changes: builder.ChangeSet,
+		State:   u.state,
+		Time:    time.Now(),
+	})
+	if err != nil {
+		return fmt.Errorf("push change: %w", err)
+	}
+	return nil
+}
+
+func (u *userDataObject) updateContactFields(profile *model.IdentityProfile, builder *storestate.Builder) error {
+	contactId := domain.NewContactId(profile.Identity)
```

Comment:
we shouldn't use identity profile here, it should be another model, which contains some fields (pls discuss with Yura), they shouldn't be taken from identity service